### PR TITLE
[New Runtime] Boot an app! (v2)

### PR DESCRIPTION
**Pull request by anurse**
We can now boot a bundled app with the new runtime!! No compilation support yet, so you have to kpm bundle --no-source and run the bundled output, but it works!

Depends on NuGet/NuGet.Packaging#18
Supersedes #1392

### CODE REVIEW

File: src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs [hunk 1]
```diff
@@ -836,6 +836,7 @@ private Task<LockFile> ReadLockFile(string projectLockFilePath)
             }
 
             // Use empty string as the key of dependencies shared by all frameworks
+
```
Comment (anurse) [inline src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs]: I'll remove this before merging.

File: src/Microsoft.Framework.Runtime.Hosting/Dependencies/ReferenceAssemblyDependencyProvider.cs [hunk 2]
```diff
@@ -31,7 +32,13 @@ public ReferenceAssemblyDependencyProvider(IFrameworkReferenceResolver framework
 
         public Library GetDescription(LibraryRange libraryRange, NuGetFramework targetFramework)
         {
-            System.Diagnostics.Debug.Assert(SupportsType(libraryRange.TypeConstraint));
+            Debug.Assert(SupportsType(libraryRange.TypeConstraint));
+
+            if(!targetFramework.IsDesktop())
```
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime.Hosting/Dependencies/ReferenceAssemblyDependencyProvider.cs]: Hmm, this isnt right either. The GAC is the only one that only works on desktop.

File: src/Microsoft.Framework.Runtime.Hosting/Internal/DefaultPackagePathResolver.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using NuGet.Versioning;
+
+namespace Microsoft.Framework.Runtime.Internal 
+{
+    public class DefaultPackagePathResolver
```
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime.Hosting/Internal/DefaultPackagePathResolver.cs]: Trim this to the minimal things needed.

File: src/Microsoft.Framework.Runtime.Hosting/Loader/PackageAssemblyLoader.cs [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    /// <summary>
+    /// Loads library assemblies from NuGet Packages
+    /// </summary>
+    /// <remarks>
+    /// This loader REQUIRES that a lock file has been generated for the project.
+    /// </remarks>
+    public class PackageAssemblyLoader : IAssemblyLoader
+    {
+        private readonly ILogger Log;
+        private Dictionary<string, string> _assemblyLookupTable = null;
```
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime.Hosting/Loader/PackageAssemblyLoader.cs]: readonly?

File: src/Microsoft.Framework.Runtime.Hosting/RuntimeHost.cs [hunk 5]
```diff
@@ -48,64 +59,80 @@ internal RuntimeHost(RuntimeHostBuilder builder)
...
+                    Project.Name,
+                    Project.Version,
+                    TargetFramework);
 
-            if (Log.IsEnabled(LogLevel.Information))
+            using (var loaderCleanup = new DisposableList())
             {
-                Log.LogInformation($"Executing Entry Point: {entryPoint.GetName().FullName}");
+                // Set up assembly loaders
+                loaderCleanup.Add(
```
Comment (davidfowl) [inline src/Microsoft.Framework.Runtime.Hosting/RuntimeHost.cs]: Make this look clearer. Split up the declaration a bit.

File: src/Microsoft.Framework.Runtime.Hosting/Dependencies/ReferenceAssemblyDependencyProvider.cs (hunk 2 above)
Comment (anurse) [inline src/Microsoft.Framework.Runtime.Hosting/Dependencies/ReferenceAssemblyDependencyProvider.cs]: Ok, I'll also update the FrameworkReferenceResolver to only rewrite DNX frameworks to Net.

File: src/Microsoft.Framework.Runtime.Hosting/Loader/PackageAssemblyLoader.cs (hunk 4 above)
Comment (anurse) [inline src/Microsoft.Framework.Runtime.Hosting/Loader/PackageAssemblyLoader.cs]: Yep, good catch.

File: src/Microsoft.Framework.Runtime.Hosting/Internal/DefaultPackagePathResolver.cs (hunk 3 above)
Comment (anurse) [inline src/Microsoft.Framework.Runtime.Hosting/Internal/DefaultPackagePathResolver.cs]: Trimming

**Pull request closed without merging**